### PR TITLE
PP-698: Update copyright year in source file headers

### PR DESCRIPTION
**Pull request by kevinlmh**
#### Issue-ID

* **[PP-698](https://pbspro.atlassian.net/browse/PP-698)**

#### Problem description
Copyright year in source file headers need to be updated from 
> Copyright (C) 1994-2016 Altair Engineering, Inc. 

to

> Copyright (C) 1994-2017 Altair Engineering, Inc.

to reflect the current year.

#### Solution description
`find`, `xargs` and `sed` were used to find the string "Copyright (C) 1994-2016" and to replace all occurrences with "Copyright (C) 1994-2017".
The full command is: `find . -type f -print0 | xargs -0 sed -i 's/Copyright (C) 1994-2016/Copyright (C) 1994-2017/g'`.

`find . -type f -print0` finds all files under the current directory (I ran this from the top level directory of pbspro). The `-print0` option handles files with spaces or tabs in their filenames. The filenames outputed by `find` are piped to `xargs` so that `sed` could be applied to each file. `sed -i 's/regex1/regex2/g' ` replaces `regex1` with `regex2`. 

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by kevinlmh** [comment]
Hello,
I am new to PBSPro so please point out things I haven't done or did wrong.
I think PTL tests are not needed for these changes since the changes only affect the comments rather than source code. Doing a successful build should be enough to verify that these changes did not break anything. 
I am still trying to figure out how to install PBS on my system and run the existing automated tests. Any help is appreciated.

**Comment by prakashcv13** [comment]
Hi @kevinlmh, 
First of all thank you very much for your contribution to PBSPro.
Answering your questions -
Yes, successful build should be enough. 
For installing PBS - I would suggest you to go through this [page](https://github.com/PBSPro/pbspro/blob/master/INSTALL). This file is also available in our source tree.
For understanding PTL, you can find the information over [here](https://pbspro.atlassian.net/wiki/display/DG/Testing+Your+Code)

**Comment by mike0042** [comment]
First, thanks for this. Now, some constructive criticism...

Files you missed (that I was able to find quickly):
- src/unsupported/NodeHealthCheck.py
- src/unsupported/pbs_rescquery.3B
- src/unsupported/run_pelog_shell.py
- LICENSE

The may be a few more, but I haven't managed to track them down yet. Your fix affects 1031 files and grep tells me there are 1042 lines with some combination of "copyright", "altair", and "1994". In some cases there may be more than one line per file. In combination with the files mentioned above that may cover everything, but I haven't confirmed it.

**Comment by kevinlmh** [comment]
Thank you @prakashcv13 and @mike0042. I will go ahead and find the other files. 
I noticed some files have inconsistent copyright statement, for example:

> Copyright (c) 2013, 2015 Altair Engineering, Inc.

in src/unsupported/run_pelog_shell.py has two years separated by comma. 

> Copyright (c) XXXX-2016 Altair Grid Technologies, LLC.

in ./src/unsupported/pbs_rescquery.3B uses Altair Grid Technologies, LLC rather than Altair Engineering, Inc. 

Should they all be changed to 

> Copyright (C) XXXX-2017 Altair Engineering, Inc.

or leaved as is? 
I do not understand the legal implications so I want to ask.

**Comment by mike0042** [comment]
@kevinlmh,

All copyright headers should be identical. The odd looking ones should be updated to match the others. Don't worry too much about the legal implications, we'll make sure the files comply before they get merged.

Please do not squash your next commit so we can see what you changed since your initial commit.

Thanks,

Mike

**Comment by kevinlmh** [comment]
@mike0042 Hi Mike. I changed a few more files I can find. I was able to build pbs on my machine but has not been able to run it so I haven't run PTL tests yet.

File: src/unsupported/pbs_rescquery.3B [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .\"
-.\" Copyright (c) 1999-2016 Altair Grid Technologies, LLC.  All rights reserved. 
+.\" Copyright (c) 1999-2017 Altair Grid Technologies, LLC.  All rights reserved.
```
Comment (mike0042) [inline src/unsupported/pbs_rescquery.3B]: The copyright notice in this file does not conform to the standard. This is not due to anything you have changed. Please remove the notice (including the paragraph below) and replace it with the notice contained in doc/man1/qsub.1B

**Comment by mike0042** [comment]
Once the test logs are attached to the JIRA ticket, you should be all set. Please squash your commits and rebase your branch. Make sure the remaining commit is signed in preparation for merge.

**Comment by kevinlmh** [comment]
@mike0042 Hi Mike. I can run individual test suites with pbs_benchpress -t <testsuitename> but not all test suite names given by pbs_benchpress -L work. Is there a quick command to run all of the existing tests at once?

**Comment by kevinlmh** [comment]
I was able to run most tests suites by putting them in a file and use `ptl_benchpress -g`. However pbs_benchpress does not recognize `TestFunctional` and `TestPerformance`.

**Comment by kjakkali** [comment]
@kevinlmh  You can use --follow-child option to run all test cases.
cmd:  pbs_benchpress --follow-child

**Comment by mike0042** [comment]
@kevinlmh please follow the advice from @kjakkali. You may squash your commits and rebase your branch.

**Comment by nithinj** [comment]
Hi @arungrover ,
Could you please merge this branch?

**Comment by mike0042** [comment]
The branch is out of date and needs updating prior to merge. This was requested on March 28th.

**Comment by kevinlmh** [comment]
Hi @mike0042 @arungrover . Sorry for the delay. I have been moving and looking for apartment. I just updated my branch.

**Comment by mike0042** [comment]
@kevinlmh Please update once more. Another change was merged since you updated.

**Comment by kevinlmh** [comment]
@mike0042 Done.

**Comment by mike0042** [comment]
Thank you @kevinlmh. Merged!

**Pull request merged**